src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,85 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.list.FunnyList;
+import duke.exception.DukeException;
+
+/**
+ *  The Storage class is responsible for handling operations with saved task data in a text file for the Duke Program.
+ *  It is capable of reading / loading and writing to the file.
+ */
+public class Storage {
+
+	private File data;
+
+	/**
+	 * Constructs a new instance of the Ui class.
+	 * Initialises an internal File object to read / write Task data.
+	 * @param filePath The path location of the data text file.
+	 */
+	public Storage(String filePath) {
+		this.data = new File(filePath);
+	}
+
+	/**
+	 * Reads the data from the text file and converts them to their corresponding task objects.
+	 * The tasks are added to an ArrayList to be returned.
+	 *
+	 * @return An ArrayList of Task objects which were originally stored in the text file.
+	 * @throws DukeException If the file specified cannot be found.
+	 */
+	public ArrayList<Task> load() throws DukeException {
+		try {
+			Scanner s = new Scanner(this.data);
+			ArrayList<Task> taskList = new ArrayList<>();
+			while (s.hasNext()) {
+				String[] row  = s.nextLine().split("\\|");
+				switch (row.length) {
```

Comment:
The `switch-case` statement needs to follow the coding standard (no indentation for the `case` clause).
Maybe also check your tab size to make sure its equal to four spaces? The indentation here seems a bit off to me.